### ISSUE TITLE
Feature request: Add option to add separators for better organization

### ISSUE THREAD

**Issue by Aleksandra0056** [aecreations/clippings#414]
I would like this option, because I want to organize my Clippings with separators.

Solution, that I'd like:
Add option to add separators between clippings and/or folders (Like the Firefox bookmarks “Add separator” option)

**Comment by aecreations** [comment]
Implement separators by adding a property in the IndexedDB database for each folder, named `seps`, which would be an array of integers representing the display order of the clipping or folder where the separator should appear immediately after it.

Example: A folder with `separators: [3, 6]` means that a separator appears after the third and sixth items in that folder.

Special case: The separator appears as the first item in the folder. In this situation, the value `-1` in the `separators` array signifies that this is the very first item in the folder, preceding the first clipping or folder (which would have display order of `0`).

But the root folder doesn't have an entry in the `folders` object store. To get around this, add a new object store named `globals` - this could be used for extension-wide exportable config options (e.g., perhaps support custom color labels in a future version). Add an entry with an object in this format: 
```
{
  name: "rootFolderSeparators",
  value: <array of integers> 
}
```
UI changes: Add a context menu item in the Clippings Manager tree list to add a new separator. The separator should be inserted immediately after the selected clipping or folder. The menu separators should be added to the submenus for each folder in the Clippings context menu.

Clippings export data (JSON format): When exporting clippings, include the `separators` property in each folder to be exported, but to keep the export data compact, the property name should be shortened to `seps`. For separators in the root folder, add the `globals` property to the top level of the export data, and use the short name `rootFldrSeps` as the config key. It should be structured as follows:
```
globals: {
  rootFldrSeps: [0, 3, 8] 
}
```
In this example, menu separators should appear after the 1st, 3rd and 8th items in the top-level Clippings menu.

This format will allow the export file to be processed by older versions of Clippings which will ignore the unsupported properties.

Importing Clippings JSON data: Process the above properties in the export JSON file, if they exist.

**NOTE:** The above is making the assumption that the display order is set on _all_ clippings and folders! This may not necessarily be the case; by default, the display order is always zero unless there is a change that causes items in the folder to be reordered. This will need to be fixed by performing first-run processing of each and every subfolder after upgrading Clippings.

**Comment by aecreations** [comment]
The disadvantage to the above approach is that it isn't possible to reorder separators using drag 'n drop in the tree list, just like the Bookmark Manager in Firefox.

Another approach is to treat separators like any ordinary tree node. This will allow drag 'n drop interactivity, and it would be possible to move, copy or delete them just like an ordinary clipping. They will also have their own `displayOrder` property in the Clippings data store, so there's no longer a need to have a `separators` property in the folder or an entirely new object store just to hold the separator positions for the root folder.

Separators will essentially be a clipping with a property `separator: true`. They can have a distinct icon, and the "name" of the separator can be something like `--- Separator ---` which is what can be displayed in Clippings Manager. Better yet, instead of rendering the node title, it could be a graphic of a horizontal line. The separator can be rendered as such when building the Clippings menu.

Disadvantage: When importing a Clippings file from version 7 or newer to an older version of Clippings, the separator will be treated as an ordinary clipping. A new option could be added to the export dialog in Clippings Manager to include or exclude separators in the export data; if separators are included in the export, they should be rendered as a clipping with the name `--- Separator ---` and its content being an empty string. Separators should not be included in the HTML or CSV export file.

**Comment by aecreations** [comment]
Separators should not be able to be moved or copied using the Move To dialog in Clippings Manager.

**Comment by aecreations** [comment]
Since separators are a new data element, but are implemented in a backward-compatible way (albeit not very gracefully, as separators will show up as a normal, oddly-named clipping in older versions), the version change on the JSON schema should be minor.

The JSON schema version in Clippings export files (*.json) and the Sync Clippings file should therefore be changed to `6.2`.

**Comment by aecreations** [comment]
MV3 extensions need to have IDs on all menu items, even separators. If a separator is missing the `id` property, it won't render.

**Comment by Aleksandra0056** [comment]
> MV3 extensions need to have IDs on all menu items, even separators. If a separator is missing the `id` property, it won't render.

Okay, I understood.

**Comment by aecreations** [comment]
Fixed in 7.0
